docs/source/federation/core-concepts.md

```diff
@@ -151,87 +193,88 @@ query {
 
-```graphql
+```graphql:title=products
 extend type Query {
   topProducts(first: Int = 5): [Product]
 }
 ```
 
-There is no need to explicitly define `Query` or `Mutation` base types anywhere; Apollo automatically handles this for you.
+## Migrating a field to another service (advanced)
 
-## Value Types
+As your federated graph grows, you might decide that you want a particular field of an entity to originate in a different service.
 
-A natural overlap among identical types between services is not uncommon. Rather than having a single service "own" those types, all services that use them are expected to share ownership. This form of type "duplication" across services is supported for Scalars, Objects, Interfaces, Enums, Unions, and Inputs. The rule of thumb for any of these value types is that the types **must be identical** in name and contents.
+For example, let's say the `products` service defines a `Product` entity that includes an `inStock` boolean field. Then, you add an `inventory` service to your federated graph. It now makes sense for the `inStock` field to originate in the `inventory` service instead.
 
-### Objects, Interfaces, and Inputs
-For types with field definitions, all fields _and their types_ must be identical.
+Apollo Gateway helps you perform this migration much like you perform a database migration, with the following steps:
 
-### Scalars
-For Scalar values, it's important that services **share the same serialization and parsing logic**, since there is no way to validate that logic from the schema level by federation tooling.
+1. In the `inventory` service, [extend](#extending) the `Product` entity to add the `inStock` field. **Note that this is technically a federation error**, because this field is already defined in the `products` service.
 
-### Enums
-For Enum types, all values must match across services. **Even if a service doesn't use all values in an Enum, they still must be defined in the schema**. Failure to include all enum values in all services that use the Enum will result in a validation error when building the federated schema.
+2. In the `inventory` service, add a resolver for the `inStock` field. This service should resolve the field with the exact same outcome as the resolver in the `products` service.
 
-### Unions
-Union types must share the same types in the union, even if not all types are used by a service.
+3. Push the updated `inventory` service to your environment. **Again, this is a federation error**. When Apollo Gateway detects a duplicate field like this, it ignores the _newer_ declaration (i.e., the one in the `inventory` service) and continues to resolve the field in the `products` service.
 
-In the following example, the Product and User services both use the same `ProductCategory` enum, `Date` scalar, `Error` type, and `ProductOrError` union.
+4. In the `products` service, remove the `inStock` field and its resolver.
```

Comment:
4 and 5 are actually out of order here. We need the gateway to stop knowing of that resolver's existence before we remove it. Otherwise, the gateway will think it's still ok to serve requests to that field (and may do so).